### PR TITLE
Obstacle avoidance for the back of the car and fix bug

### LINKED ISSUE #34: The car avoids obstacles that are behind it while being manually maneuvered.
### User Story

As a medical worker, I want the car to avoid obstacles behind it, so that the car doesn't get damaged.

### Acceptance Criteria

- The car should avoid obstacles that are about 30cm behind it.
- There should be a popup notification when an obstacle has been avoided

### PR DESCRIPTION
**Pull request by Marble879**
### Changes

1. Add obstacle avoidance for when maneuvering the car backwards.
2. Reduce the max speed of the car from 100% to 60% so that it is better suited for maneuvering in a hospital.
3. Fix bug where the Arduino sketch wouldn't compile after the SMCE 1.2.1 update.

### Issues
Resolves #34

### CODE REVIEW

**Review by NilsDunlop — APPROVED** [review]
After have tested the back speed values in the SMCE they seem to make the IRSensors identify obstacles properly as compared to before. Other than that, the code looks readable and good. Well done!

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -86,14 +86,18 @@ void loop() {
...
+  
+  boolean isFrontDetected = frontDistanceFromObject < stopDistanceFront && frontDistanceFromObject > 1 && !(carSpeed <= 0);
+  boolean isBackDetected = backDistanceFromObject < stopDistanceBack && backDistanceFromObject > 1 && !(carSpeed >= 0);
+  
+  if (isFrontDetected || isBackDetected){
+    car.setSpeed(stoppingSpeed);
+    return true;
+  } else {
+    return false;  
+  }
```
Comment (MarkusJ299) [inline arduino/smartcar/smartcar.ino]: Would it be better to use another boolean variable that is returned in this method? As a result, only one return statement would be needed and the else-condition would no longer be necessary.

```suggestion
  boolean isObstacleDetected = false;
  
  if (isFrontDetected || isBackDetected){
    car.setSpeed(stoppingSpeed);
    isObstacleDetected = true;
  }
  
  return isObstacleDetected;
```

**Review by MarkusJ299 — CHANGES_REQUESTED** [review]
Looks good overall. I just added a minor suggestion. Let me know what you think.

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Marble879) [reply arduino/smartcar/smartcar.ino]: That's a great point! It improves the readability of the code - I'll add the changes now :+1:

**Review by MarkusJ299 — APPROVED** [review]
Looks good!

**Review by puerto-hub — APPROVED** [review]
Seems good! Great work by making two obstacle detection booleans.

**Review by JihanP — APPROVED** [review]
It looks good, nice that you restricted the speed for the car! Good job, keep going!

**Review by Kevinayad — APPROVED** [review]
Good job!

**Review by gusasino — COMMENTED** [review]
great work!

**Pull request merged**